utils/io/file.go

```diff
@@ -84,28 +86,67 @@ func CopyDirectory(scrDir, dest string) error {
 	return nil
 }
 
-func Copy(srcFile, dstFile string) error {
-	out, err := os.Create(dstFile)
+func Copy(srcFile, dstFile string) (errToReturn error) {
+	in, err := os.Open(srcFile)
 	if err != nil {
-		return err
+		return fmt.Errorf("can not open file %v to copy from: %w", srcFile, err)
 	}
 
-	defer out.Close()
+	defer func() {
+		errToReturn = CloseAndMergeError(in, errToReturn)
+	}()
```

Comment:
Uber's version on multierr suggests using a simpler version by using the [AppendInvoke](https://pkg.go.dev/go.uber.org/multierr#AppendInvoke):

```
defer multierr.AppendInvoke(&err, multierr.Close(in))
```